pytorch_lightning/utilities/auto_restart.py

```diff
@@ -116,12 +116,13 @@ def _compute_current_iteration(self, num_batches_processed: Optional[int] = None
 
         return current_iteration
 
-    def _load_cached_state(self):
+    def _load_cached_state(self) -> None:
         if self._cached_state_dict is None or self.worker_id not in self._cached_state_dict:
             return
         self._current_iteration = self._cached_state_dict[self.worker_id]["current_iteration"]
         # delete cached state, prevent reloading every time iter() is called
         self._cached_state_dict = None
+        return
```

Comment:
pretty sure this is not necessary :/

Reply:
should we put `warn_no_return = False` for mypy?

Reply:
I remember I read a PEP recommendation suggesting if there's any `return` statement within a function, it should be obeyed for the whole function definition. But yes, my suggestion might be redundant here.